### ISSUE TITLE
Missing parameter annotations if I don't load class

### ISSUE THREAD

**Issue by dantesun** [lukehutch/fast-classpath-scanner#151]
As the title described. It seems that the parameter annotation is missing from MethodInfo.

**Comment by lukehutch** [comment]
@dantesun I don't understand the description, can you please give me an example?

**Comment by dantesun** [comment]
```
@Retention(RetentionPolicy.CLASS)
@Target(ElementType.PARAMETER)
public @interface ParamAnnotation {
}
```

`public void foo(@ParamAnnotation String annotatedValue)`

I mean the annotation in parameter declaration should also included in MethodInfo class

**Comment by lukehutch** [comment]
@dantesun Fixed in master, but to reduce version churn, I might wait a bit before pushing out another version. Are there other changes you think you will need anytime soon?

@dantesun I think you are the only one using method parameter info in the API at this point. The API currently returns a list for each of the parameter info types (parameter names, parameter modifiers, and parameter attributes). However, the number of parameters is fixed, so maybe I should return an array rather than a list? Do you care which type is returned?

**Comment by dantesun** [comment]
@lukehutch Thank you very much for the quick fix. Parameter annotations are widely used in Spring MVC framework. I am doing some kind of analysis for statically scanning all the Spring MVC controller like class. So technically everything in a method signature matters to me.

**Comment by lukehutch** [comment]
@dantesun makes sense. Please carefully check if anything else is missing, and I'll push out a release once you think it's complete.

PS please see my question above about array vs. list types for parameter introspection.

**Comment by dantesun** [comment]
@lukehutch I have no concern about the type returned, list or array both are good to me. I have created a value class for method parameter, such as:
```
@Immutable
public interface JavaMethodParameter {

    /**
     * All the annotations presented on this parameter
     *
     * @return a immutable list of full class name of annotation
     */
    List<String> getAnnotations();

    /**
     * The declaring position in a method
     *
     * @return index, starts with 0
     */
    int getIndex();

    /**
     * <p>
     * The name defined in source code. See https://docs.oracle
     * .com/javase/tutorial/reflect/member/methodparameterreflection.html for details.
     * </p>
     *
     * @return true if using JDK8 "MethodParameter"
     */
    boolean isNameDefined();

    /**
     * The name of a method parameter
     *
     * @return user defined name or "arg${index}" if not compiled with "-parameter"
     * @see #isNameDefined()
     */
    String getName();
//... ignored
```

**Comment by lukehutch** [comment]
Fixed and released in version 2.8.2.

**Comment by lukehutch** [comment]
@dantesun I changed parameter annotations back to use arrays rather than lists for method parameter annotations in the latest release (2.9.0), to avoid unnecessarily creating a `List` wrapper object. Sorry for the API churn. Version 2.9.0 also includes support for parameter annotation values, which is a major new change -- please see the release notes
